### PR TITLE
feat(inverse_transform): enable fit and transform with horizontal_matrix

### PR DESCRIPTION
**Pull request by raimbaultL**
After experiments, we conclude that allowing avatarization with a horizontal matrix is not a problem.

### CODE REVIEW

**Comment by mguillaudeux** [comment]
**Summary of Dimensionality Limitation in PCA, MCA, and FAMD**

The maximum number of dimensions in methods like PCA (ACP), MCA (ACM), and FAMD (AFDM) is constrained by the rank of the data matrix. This rank is equal to the minimum of the number of rows (n, individuals) and columns (p, variables).

    In PCA, the covariance matrix's rank cannot exceed the smallest of n or p, meaning the number of principal components is limited by this minimum.
    In MCA, working with qualitative data, the number of factor axes is similarly limited by the degrees of freedom in the matrix.
    In FAMD, which handles mixed data, the same logic applies: the number of dimensions is restricted by the matrix rank, reflecting the minimum between the number of individuals and variables.

Example: For a dataset with 100 individuals and 50 variables, the maximum number of dimensions after projection will be 50, as there are only 50 variables to define the variability.

This constraint is a fundamental mathematical property of matrix rank, which determines the number of independent linear combinations available from the data.

Technically other components could be added but with 0 more variance explained as they would necessarily be ortogonal with an already existing one.

Please Note that 100% of total variance is still captured in this limited number of components

File: saiph/projection_test.py [hunk 1]
```diff
@@ -540,3 +540,30 @@ def test_transform_raise_error_on_wrong_columns(
...
+            }
+        ),  # mca
+    ],
+)
+def test_fit_transform_with_horizontal_matrix(
+    df_to_fit_transform: pd.DataFrame,
+) -> None:
+    """Verify that the coordinates are a squared matrix even if the input is horizontal."""
+    coord, __ = fit_transform(df_to_fit_transform)
+    assert coord.shape[0] == coord.shape[0]
```
Comment (mguillaudeux) [inline saiph/projection_test.py:569]: glad you added this test

**Review by mguillaudeux — APPROVED** [review]
LGTM ! 
I added some theorical arguments in the Conversation of the PR.
I also wonder what we should do when the user purposly specifies nf > min(n, p) in the fit function? Error, Warning, Noting ?
Personnally I would like to avoid retruning to much errors when not necessary

**Comment by albanfelix** [comment]
> **Summary of Dimensionality Limitation in PCA, MCA, and FAMD**
> 
> The maximum number of dimensions in methods like PCA (ACP), MCA (ACM), and FAMD (AFDM) is constrained by the rank of the data matrix. This rank is equal to the minimum of the number of rows (n, individuals) and columns (p, variables).
> 
> ```
> In PCA, the covariance matrix's rank cannot exceed the smallest of n or p, meaning the number of principal components is limited by this minimum.
> In MCA, working with qualitative data, the number of factor axes is similarly limited by the degrees of freedom in the matrix.
> In FAMD, which handles mixed data, the same logic applies: the number of dimensions is restricted by the matrix rank, reflecting the minimum between the number of individuals and variables.
> ```
> 
> Example: For a dataset with 100 individuals and 50 variables, the maximum number of dimensions after projection will be 50, as there are only 50 variables to define the variability.
> 
> This constraint is a fundamental mathematical property of matrix rank, which determines the number of independent linear combinations available from the data.
> 
> Technically other components could be added but with 0 more variance explained as they would necessarily be ortogonal with an already existing one.
> 
> Please Note that 100% of total variance is still captured in this limited number of components

The rank is not equal to the minimum dimension, but **bounded** by the minimal dimension. If rank(matrix)=min(n,p), the matrix is said to be full ranked.

File: saiph/inverse_transform.py [hunk 2]
```diff
@@ -36,16 +32,7 @@ def inverse_transform(
...
 
-    # Check dimension size regarding N
-    n_dimensions = len(model.dummy_categorical) + len(model.original_continuous)
-    n_records = len(coord)
-
-    if not use_approximate_inverse and n_records < n_dimensions:
-        raise InvalidParameterException(
-            f"n_dimensions ({n_dimensions}) is greater than n_records ({n_records})."
-        )
     # Get back scaled_values from coord with inverse matrix operation
```
Comment (jpetot) [inline saiph/inverse_transform.py:35]: I don't understand what is the approximate inverse if this is the only place where we used it

**Review by jpetot — APPROVED** [review]
The changes look good to me. I wonder what the approximate inverse is, or how the inverse transformation works when the data frame is horizontal.

File: saiph/inverse_transform.py (hunk 2 above)
Comment (raimbaultL) [reply saiph/inverse_transform.py:35]: The inverse transform can't be in horizontal matrix as the coordinates are at worst squared. 
and we use https://numpy.org/doc/stable/reference/generated/numpy.linalg.pinv.html to do the inverse as https://numpy.org/doc/stable/reference/generated/numpy.linalg.inv.html
is only on squared matrix

**Pull request merged**